can/typechecking.py

```diff
@@ -8,10 +8,11 @@
 
 import mypy_extensions
 
-CanFilter = mypy_extensions.TypedDict(
-    "CanFilter", {"can_id": int, "can_mask": int, "extended": bool}
+CanFilter = mypy_extensions.TypedDict("CanFilter", {"can_id": int, "can_mask": int})
```

Comment:
Is this correct? I'd have thought both `CanFilter` and `CanFilterExtended` have an `extended` bool attribute - just one is True and one is False?

Reply:
I would've expected that also, but currently in the code we also accept cases where the `extended` key is not present. The SocketCAN interface is one such place:

```python
def pack_filters(can_filters=None):
    if can_filters is None:
        # Pass all messages
        can_filters = [{"can_id": 0, "can_mask": 0}] # <---- here

    can_filter_fmt = "={}I".format(2 * len(can_filters))
    filter_data = []
    for can_filter in can_filters:
        can_id = can_filter["can_id"]
        can_mask = can_filter["can_mask"]
        if "extended" in can_filter:
            # Match on either 11-bit OR 29-bit messages instead of both
            can_mask |= CAN_EFF_FLAG
            if can_filter["extended"]:
                can_id |= CAN_EFF_FLAG
        filter_data.append(can_id)
        filter_data.append(can_mask)

    return struct.pack(can_filter_fmt, *filter_data)
```

Reply:
ok thanks for clarifying for me